modules/services/repositories/src/main/java/au/com/shiftyjelly/pocketcasts/repositories/bookmark/BookmarkHelper.kt

```diff
@@ -0,0 +1,117 @@
+package au.com.shiftyjelly.pocketcasts.repositories.bookmark
+
+import android.Manifest
+import android.app.PendingIntent
+import android.content.Context
+import android.content.Intent
+import android.content.pm.PackageManager
+import androidx.core.app.ActivityCompat
+import androidx.core.app.NotificationCompat
+import androidx.core.app.NotificationManagerCompat
+import au.com.shiftyjelly.pocketcasts.preferences.Settings
+import au.com.shiftyjelly.pocketcasts.preferences.Settings.Companion.INTENT_OPEN_APP_ADD_BOOKMARK
+import au.com.shiftyjelly.pocketcasts.preferences.Settings.Companion.INTENT_OPEN_APP_VIEW_BOOKMARKS
+import au.com.shiftyjelly.pocketcasts.repositories.playback.PlaybackManager
+import au.com.shiftyjelly.pocketcasts.utils.extensions.isAppForeground
+import au.com.shiftyjelly.pocketcasts.utils.log.LogBuffer
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+import kotlin.coroutines.CoroutineContext
+import au.com.shiftyjelly.pocketcasts.images.R as IR
+import au.com.shiftyjelly.pocketcasts.localization.R as LR
+
+class BookmarkHelper @Inject constructor(
+    private val playbackManager: PlaybackManager,
+    private val bookmarkManager: BookmarkManager,
+    private val settings: Settings,
+) : CoroutineScope {
+    override val coroutineContext: CoroutineContext
+        get() = Dispatchers.Default
+
+    fun handleAddBookmarkAction(
+        context: Context,
+    ) {
+        if (context.isAppForeground()) {
+            val bookmarkIntent =
+                context.packageManager.getLaunchIntentForPackage(context.packageName)
+                    ?.apply { action = INTENT_OPEN_APP_ADD_BOOKMARK }
+            bookmarkIntent?.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
+            context.startActivity(bookmarkIntent)
+        } else {
```

Comment:
I'm not sure about this. Maybe we don't need this if the bookmark is added using headset controls 🤔